openedx/core/lib/api/view_utils.py

```diff
@@ -0,0 +1,28 @@
+"""
+Utilities related to API views
+"""
+from django.http import Http404
+
+from rest_framework.exceptions import APIException
+from rest_framework.response import Response
+
+
+class DeveloperErrorViewMixin(object):
+    """
+    A view mixin to handle common error cases other than validation failure
+    (auth failure, method not allowed, etc.) by generating an error response
+    conforming to our API conventions with a developer message.
+    """
+    def make_error_response(self, status_code, developer_message):
+        """
+        Build an error response with the given status code and developer_message
+        """
+        return Response({"developer_message": developer_message}, status=status_code)
+
+    def handle_exception(self, exc):
+        if isinstance(exc, APIException):
+            return self.make_error_response(exc.status_code, exc.detail)
+        elif isinstance(exc, Http404):
```

Comment:
Thanks for pointing that out. In general for our REST APIs, I think it makes sense to use DRF's Response object so DRF can return the requested type of content even in the 404 case.